### PR TITLE
COMMERCE-1496 CommerceOrder replace commerceCurrencyId with currencyCode

### PR DESCRIPTION
**Pull request by luca-pellizzon**

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test

File: commerce-service/src/main/java/com/liferay/commerce/model/impl/CommerceOrderItemImpl.java [hunk 1]
```diff
@@ -69,32 +70,40 @@ public CProduct getCProduct() throws PortalException {
 	public CommerceMoney getDiscountAmountMoney() throws PortalException {
 		CommerceOrder commerceOrder = getCommerceOrder();
 
+		CommerceCurrency commerceCurrency = commerceOrder.getCommerceCurrency();
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/model/impl/CommerceOrderItemImpl.java:73]: @luca-pellizzon We have to refactor Commerce Money and commerce Price to use the code as well

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Comment by luca-pellizzon** [comment]
@marco-leo @ethanbustad 
PR updated

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
@marco-leo check this pls

**Comment by luca-pellizzon** [comment]
@ethanbustad upgrade process updated here like https://github.com/liferay/com-liferay-commerce/pull/1645

File: commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v5_0_0/CommerceOrderUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import com.liferay.portal.kernel.util.StringBundler;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Luca Pellizzon
+ */
+public class CommerceOrderUpgradeProcess extends UpgradeProcess {
```
Comment (ethanbustad) [inline commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v5_0_0/CommerceOrderUpgradeProcess.java:32]: If you extend `BaseCommerceServiceUpgradeProcess`, you don't have to copy over `_addColumn` and `_dropColumn` :)

File: commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v5_0_0/CommerceOrderUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+		try {
+			ps = connection.prepareStatement(
+				"update CommerceOrder set commerceCurrencyCode = ? where " +
+					"commerceCurrencyId = ?");
+
+			s = connection.createStatement();
+
+			rs = s.executeQuery(
+				"select commerceCurrencyId, code_ from CommerceCurrency");
```
Comment (ethanbustad) [inline commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v5_0_0/CommerceOrderUpgradeProcess.java:59]: This would be nicer as a try-with-resources, if you don't mind. It saves several lines, and prevents us from having to remember to clean up at the end.

File: commerce-service/service.xml [hunk 4]
```diff
@@ -249,7 +249,7 @@
 		<!-- Other fields -->
 
 		<column name="commerceAccountId" type="long" />
-		<column name="commerceCurrencyId" type="long" />
+		<column name="commerceCurrencyCode" type="String" />
```
Comment (ethanbustad) [inline commerce-service/service.xml:252]: this should be sorted below `shippingAddressId` now (at least that low -- put it where you want it after that)

File: commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/servlet/CommercePaymentMethodPayPalServlet.java [hunk 5]
```diff
@@ -93,7 +94,7 @@ protected void doGet(
 				String token = ParamUtil.getString(httpServletRequest, "token");
 
 				if (paymentId.isEmpty() && !token.isEmpty()) {
-					_commercePaymentEngine.completeRecurringPayment(
+					_commerceSubscriptionEngine.completeRecurringPayment(
```
Comment (ethanbustad) [inline commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/servlet/CommercePaymentMethodPayPalServlet.java:97]: Is this related? :)

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java [hunk 6]
```diff
@@ -203,7 +203,9 @@ public CommerceOrder addCommerceOrder(
 
 		validateGuestOrders();
 
-		if (Validator.isNull(commerceCurrencyCode)) {
+		if (Validator.isNull(commerceCurrencyCode) ||
+			Validator.isBlank(commerceCurrencyCode)) {
```
Comment (ethanbustad) [inline commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java:207]: This... shouldn't be necessary.
`Validator.isNull` checks for the string being `null`, `""`, or `"null"`.
`Validator.isBlank` checks for the string being `null` or `""`.
So `isNull` is a strict superset of `isBlank`, so we should never have to check both.

File: commerce-test-util/src/main/java/com/liferay/commerce/test/util/CommerceTestUtil.java [hunk 7]
```diff
@@ -72,11 +72,11 @@ public static CommerceOrder addB2BCommerceOrder(
 			userId = serviceContext.getUserId();
 		}
 
-		if (commerceCurrencyId == 0) {
+		if (commerceCurrencyCode.isEmpty()) {
```
Comment (ethanbustad) [inline commerce-test-util/src/main/java/com/liferay/commerce/test/util/CommerceTestUtil.java:75]: should probably use `Validator.isBlank`

File: commerce-test-util/src/main/java/com/liferay/commerce/test/util/CommerceTestUtil.java [hunk 8]
```diff
@@ -99,11 +99,11 @@ public static CommerceOrder addB2CCommerceOrder(
...
```
Comment (ethanbustad) [inline commerce-test-util/src/main/java/com/liferay/commerce/test/util/CommerceTestUtil.java:102]: `Validator.isBlank`

**Review by ethanbustad — CHANGES_REQUESTED** [review]

**Comment by marco-leo** [comment]
This cahnges will be adressed by ALec : https://github.com/liferay/com-liferay-commerce/pull/1612
